### PR TITLE
Collaboration service real time

### PR DESCRIPTION
**Pull request by gycgabriel**
- Operational Transformations in Collaboration Service
- Add utils/shared-ot.ts
  - next.config.js: Enable experimental feature - external dir
  - Some scary warning appears here
- Fix frontend
  - remove debouncer
- Add cursor support, defaults to 0

Current design:

Client --> Server: OT + Cursor
Server --> Client: Full text + Version number + Cursor

Version number incremented by Server, client should not increment because version number refers to previous version that is edited upon.

Client waits for: 
- Acknowledgement from server of prev request
- Sync version (and text) from server
before they can send the next update request

This ensures no conflicting versions and mix-ups.

### CODE REVIEW

**Review by ong6 — APPROVED** [review]
LGTM - Excellent work on this, the only concern now is the user cursor in the FE since after the text is set the cursor will default to the end of the text.

**Comment by gycgabriel** [comment]
- Added cursor support
- Move Rooms db to Prisma
- Fix mergeOp to multiple versions from memory if previous version is very old for proper transforming

**Comment by ong6** [comment]
Lets merge after the demo tonight - to play safe

File: frontend/src/hooks/useCollaboration.tsx [hunk 1]
```diff
@@ -1,30 +1,71 @@
...
+        textRef.current = text;
+        prevTextRef.current = text;
         setText(text);
-      });
-    }
+        if (cursor && cursor > -1) {
+          console.log("Update cursor to " + cursor);
+          cursorRef.current = cursor;
+          setCursor(cursor);
+        }
```
Comment (gycgabriel) [inline frontend/src/hooks/useCollaboration.tsx:76]: I'm not sure why this doesn't set the cursor, the log message shows the correct cursor position

Comment (gycgabriel) [reply frontend/src/hooks/useCollaboration.tsx:76]: @ckcherry23 @ong6 🙏

**Pull request merged**